examples/all-clusters-app/esp32/main/main.cpp

```diff
@@ -495,119 +493,6 @@ void SetupPretendDevices()
     AddAttribute("State", "Closed");
 }
 
-void GetGatewayIP(char * ip_buf, size_t ip_len)
-{
-    esp_netif_ip_info_t ipInfo;
-    esp_netif_get_ip_info(esp_netif_get_handle_from_ifkey("WIFI_AP_DEF"), &ipInfo);
-    esp_ip4addr_ntoa(&ipInfo.ip, ip_buf, ip_len);
-    ESP_LOGI(TAG, "Got gateway ip %s", ip_buf);
-}
-
-bool isRendezvousBLE()
-{
-    RendezvousInformationFlags flags = RendezvousInformationFlags(CONFIG_RENDEZVOUS_MODE);
-    return flags.Has(RendezvousInformationFlag::kBLE);
-}
-
-std::string createSetupPayload()
-{
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    std::string result;
-
-    uint16_t discriminator;
-    err = ConfigurationMgr().GetSetupDiscriminator(discriminator);
-    if (err != CHIP_NO_ERROR)
-    {
-        ESP_LOGE(TAG, "Couldn't get discriminator: %s", ErrorStr(err));
-        return result;
-    }
-    ESP_LOGI(TAG, "Setup discriminator: %u (0x%x)", discriminator, discriminator);
-
-    uint32_t setupPINCode;
-    err = ConfigurationMgr().GetSetupPinCode(setupPINCode);
-    if (err != CHIP_NO_ERROR)
-    {
-        ESP_LOGE(TAG, "Couldn't get setupPINCode: %s", ErrorStr(err));
-        return result;
-    }
-    ESP_LOGI(TAG, "Setup PIN code: %u (0x%x)", setupPINCode, setupPINCode);
-
-    uint16_t vendorId;
-    err = ConfigurationMgr().GetVendorId(vendorId);
-    if (err != CHIP_NO_ERROR)
-    {
-        ESP_LOGE(TAG, "Couldn't get vendorId: %s", ErrorStr(err));
-        return result;
-    }
-
-    uint16_t productId;
-    err = ConfigurationMgr().GetProductId(productId);
-    if (err != CHIP_NO_ERROR)
-    {
-        ESP_LOGE(TAG, "Couldn't get productId: %s", ErrorStr(err));
-        return result;
-    }
-
-    SetupPayload payload;
-    payload.version               = 0;
-    payload.discriminator         = discriminator;
-    payload.setUpPINCode          = setupPINCode;
-    payload.rendezvousInformation = RendezvousInformationFlags(CONFIG_RENDEZVOUS_MODE);
-    payload.vendorID              = vendorId;
-    payload.productID             = productId;
-
-    if (!isRendezvousBLE())
-    {
-        char gw_ip[INET6_ADDRSTRLEN];
-        GetGatewayIP(gw_ip, sizeof(gw_ip));
```

Comment:
The previous code added this optional element in case of Rendezvous SoftAP. Do we still need that?

Reply:
It should be handled by the common `OnboardingCodesUtil.cpp`, although the SoftAP is not well supported right now.